src/test/scala/com/fulcrumgenomics/bam/EstimateRnaSeqInsertSizeTest.scala

```diff
@@ -0,0 +1,192 @@
+    builder.addPair(contig=2, start1=1, start2=133801671) // before ACKR4-3
+    builder.addPair(contig=2, start1=133801671, start2=133814175) // after ACKR4-3
+    val bam = builder.toTempFile()
+    val out = PathUtil.pathTo(PathUtil.removeExtension(bam) + EstimateRnaSeqInsertSize.RnaSeqInsertSizeMetricExtension)
+    new EstimateRnaSeqInsertSize(input=bam, refFlat=RefFlatFile, minimumOverlap=0.0).execute()
+    val metrics = Metric.read[InsertSizeMetric](path=out)
+    metrics.length shouldBe PairOrientation.values().length
+    metrics.zip(EmptyMetrics).foreach { case (actual, expected) =>
+      actual shouldBe expected
+    }
+  }
+
+  it should "ignore reads when the insert size is different across transcripts" in {
+    val builder = new SamRecordSetBuilder()
+    builder.addPair(contig=4, start1=133801671, start2=133804077) // overlaps ACKR4-5 (multiple transcripts)
+    val bam = builder.toTempFile()
+    val out = PathUtil.pathTo(PathUtil.removeExtension(bam) + EstimateRnaSeqInsertSize.RnaSeqInsertSizeMetricExtension)
+    new EstimateRnaSeqInsertSize(input=bam, refFlat=RefFlatFile, minimumOverlap=0.0).execute()
+    val metrics = Metric.read[InsertSizeMetric](path=out)
+    metrics.length shouldBe PairOrientation.values().length
+    metrics.zip(EmptyMetrics).foreach { case (actual, expected) =>
+      actual shouldBe expected
+    }
+  }
+
+  it should "ignore reads when there are too few mapped bases overlapping exonic sequence" in {
+    val builder = new SamRecordSetBuilder()
+    builder.addPair(contig=5, start1=133801671, start2=133804076, strand1=Plus, strand2=Plus) // overlaps ACKR4-6 by 2 bases!
+    val bam = builder.toTempFile()
+    val out = PathUtil.pathTo(PathUtil.removeExtension(bam) + EstimateRnaSeqInsertSize.RnaSeqInsertSizeMetricExtension)
+
+    // OK
+    {
+      new EstimateRnaSeqInsertSize(input=bam, refFlat=RefFlatFile, minimumOverlap=2/200.0).execute()
+      val metrics = Metric.read[InsertSizeMetric](path=out)
+      metrics.length shouldBe PairOrientation.values().length
+      metrics.zip(EmptyMetrics).foreach { case (actual, expected) =>
+        if (actual.pairOrientation == PairOrientation.TANDEM) {
+          actual shouldBe InsertSizeMetric(
+            pairOrientation = PairOrientation.TANDEM,
+            read_pairs         = 1,
+            mean               = 1,
+            min                = 1,
+            max                = 1,
+            median             = 1
+          )
+        }
+        else {
+          actual shouldBe expected
+        }
+      }
+    }
+
+    // Too few bases overlapping exonic sequence
+    {
+      new EstimateRnaSeqInsertSize(input=bam, refFlat=RefFlatFile, minimumOverlap=3/200.0).execute()
+      val metrics = Metric.read[InsertSizeMetric](path=out)
+      metrics.length shouldBe PairOrientation.values().length
+      metrics.zip(EmptyMetrics).foreach { case (actual, expected) =>
+        actual shouldBe expected
+      }
+    }
+  }
+}
```

Comment:
I totally agree about the tests, I'll try and fix them at a later date as I did some refactoring that should make it easier.